start.py

```diff
@@ -259,7 +276,46 @@ def _remove_files(nodes, files, quiet):
         node.execute(command=command, quiet=quiet)
 
 
+def _install_atlas_streamsets_model(node, custom_stage_lib, quiet):
+    logger.debug('Installing StreamSets Atlas model ...')
+    atlas_custom_stage_lib_name, atlas_custom_stage_lib_version = custom_stage_lib.split(',')
+    atlas_custom_stage_lib_version = ('{}-SNAPSHOT'.format(atlas_custom_stage_lib_version.rsplit('-latest', 1)[0])
+                                      if atlas_custom_stage_lib_version.endswith('-latest')
+                                      else atlas_custom_stage_lib_version)
+    image_name = ATLAS_STAGE_IMAGE_NAME_TEMPLATE.format(atlas_custom_stage_lib_name, atlas_custom_stage_lib_version)
+    try:
+        docker_client.api.inspect_image(image_name)
+    except docker.errors.NotFound as not_found:
+        if (not_found.response.status_code == 404 and 'No such image' in not_found.explanation):
+            docker_client.images.pull(image_name)
+
+    file_path = ('/opt/streamsets-datacollector-user-libs/streamsets-datacollector-{}-lib/lib/'
+                 'streamsets-datacollector-apache-atlas-protolib-{}.jar'.format(atlas_custom_stage_lib_name,
+                                                                                atlas_custom_stage_lib_version))
+    container_id = docker_client.api.create_container(image=image_name)['Id']
+    try:
+        container = docker_client.containers.get(container_id=container_id)
+
+        tarstream = io.BytesIO()
+        for chunk in container.get_archive(path=file_path)[0]:
+            tarstream.write(chunk)
+        tarstream.seek(0)
+
+        zfile = zipfile.ZipFile(tarstream)
```

Comment:
Could you add a comment about what's going on here? It's just a bit hard to follow.